google/cloud/spanner_dbapi/cursor.py

```diff
@@ -258,9 +260,56 @@ def executemany(self, operation, seq_of_params):
 
         many_result_set = StreamedManyResultSets()
 
-        for params in seq_of_params:
-            self.execute(operation, params)
-            many_result_set.add_iter(self._itr)
+        if classification in (parse_utils.STMT_INSERT, parse_utils.STMT_UPDATING):
+            statements = []
+
+            for params in seq_of_params:
+                sql, params = parse_utils.sql_pyformat_args_to_spanner(
+                    operation, params
+                )
+                statements.append((sql, params, get_param_types(params)))
+
+            if self.connection.autocommit:
+                transaction = self.connection.transaction_checkout()
+                status, res = transaction.batch_update(statements)
+                many_result_set.add_iter(res)
+
+                if status.code != OK:
+                    self.connection._transaction = None
```

Comment:
You should explicitly rollback the transaction here. The transaction will be aborted by Cloud Spanner after 10 seconds if you just keep it open here, but that would mean that it will unnecessarily hold on to locks for 10 seconds.

(This comment is void if we use `run_in_transaction`, as that will handle the transaction for you.)

Reply:
Done